src/main/webapp/assets/js/feedScript.js

```diff
@@ -933,3 +1006,115 @@ function toggleLegend() {
+  addPosts(filteredPosts);
+  addMarkers(filteredPosts);
+}
+
+/**
+ * Applies the filtering algorithm to an array of posts.
+ * The flow is taken from functional programming principles, in which each new filter is applied
+ * to the result of the last one. For each filter, we only apply it if the filter's state is
+ * non-default, i.e. the user has interacted with it to change it from the default and is interested
+ * in filtering on that dimension.
+ * @param {Array<PostInfo>} posts - The posts to filter.
+ * @param {Date} now - The current datetime.
+ * @param {Object} userLocation - The user's location (lat/long).
+ * @param {Filter} filters - The parameters by which to filter.
+ * @return {Array <PostInfo>} - The filtered list of posts.
+ */
+function filterPosts(posts, now, userLocation, filters) {
+  let filteredPosts = posts;
+  if (filters.numPeople !== NUM_PEOPLE_FILTER_DEFAULT) {
+    filteredPosts = filteredPosts.filter((p) => p.numOfPeopleFoodWillFeed >= filters.numPeople);
+  }
+  if (filters.happeningNow !== HAPPENING_NOW_FILTER_DEFAULT) {
+    filteredPosts = filteredPosts.filter((p) => {
+      return now >= p.eventStartTime && now <= p.eventEndTime;
+    });
+  }
+  if (filters.distance !== DISTANCE_FILTER_DEFAULT && userLocation) {
+    filteredPosts = filteredPosts.filter((p) => {
+      return calculateMilesBetweenTwoCoords(p.location, userLocation) <= filters.distance;
+    });
+  }
+  if (filters.keywords !== KEYWORDS_FILTER_DEFAULT && filters.keywords.length > 0) {
+    filteredPosts = filteredPosts.filter((p) => {
+      const combinedText = `${p.organizationName} ${p.foodType} ${p.description}`;
+      for (const keyword of filters.keywords) {
+        if (combinedText.toLowerCase().includes(keyword.toLowerCase())) {
+          return true;
+        }
+      }
+      return false;
+    });
+  }
+
+  return filteredPosts;
+}
+
+/**
+ * Calculates the distance (in miles) between two geographic coordinates.
+ * @param {Object} locationA - The first location.
+ * @param {Object} locationB - The second location.
+ * @return {number} - The distance between locationA and locationB in miles.
+ */
+function calculateMilesBetweenTwoCoords(locationA, locationB) {
+  // Find the average latitude so we can determine miles/longitude°, since that distance varies
+  // from pole to equator. Miles/latitude°, on the other hand, is relatively constant.
+  // (See https://gis.stackexchange.com/questions/142326/calculating-longitude-length-in-miles).
+  const avgLat = (locationA.lat + locationB.lat)/2;
+  const milesPerDegreeLong = Math.cos(avgLat * Math.PI / 180) * MILES_PER_DEGREE_LAT;
+
+  // Multiple each absolute difference (in degrees) by miles/° for each dimension.
+  const latDiff = Math.abs(locationA.lat - locationB.lat) * MILES_PER_DEGREE_LAT;
+  const longDiff = Math.abs(locationA.long - locationB.long) * milesPerDegreeLong;
+
+  // Use the Pythagorean theorem to find the hypotenuse length given latDiff and longDiff
```

Comment:
I'd thought I would never see this again :000

Reply:
Turns out my teachers were right all along, this would be used in the real world haha